src/controllers/company.number.controller.ts

```diff
@@ -47,7 +47,7 @@ const formatCompanyNumber = (companyNumber: string, leadPoint: number): string
 // validator middleware that checks for invalid characters in the input
 const postValidators = [
   check(companyNumberFieldName).blacklist(" ").escape().custom((value: string) => {
-    if (value.length !== 8 || !/^[a-z]{0,2}?\d+?$/gmi.test(value)) {
+    if (value.length !== 8 || !/^(?:[a-z]|[a-z][a-z])?\d{6,8}?$/gmi.test(value)) {
```

Comment:
?:  is a non capturing group, without it the group () will be captured so we can use it later on, but we don't need to capture it, hence added ?:

Reply:
Sonar doesn't like 2 or more of +*{ so i've made the 1 or 2 letters at start literal rather then use the {0,2} occurrences, so we can use {} or + after the d. I've gone for {6,8} to narrow it down a bit, + would allow infinite number of digits. We have a length check in the code before this, so any string being evaluated will always be length of 8.